### PR TITLE
[14.0][MIG] mrp_subcontracting_skip_no_negative: Migration to 14.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 14.0

@Tecnativa TT50668

### CODE REVIEW

**Review by gbrito — APPROVED** [review]

**Review by pedrobaeza — COMMENTED** [review]
Reading the code, I'm not sure there will be an error when the material to consume is not available. The test also doesn't seem to test a real subcontracting flow. What you confirm is the PO receipt, not other manual picking that I see in the test.

**Comment by victoralmau** [comment]
Sorry, I don't understand exactly what you indicate (what I am forgetting).

Use case (test and UI).

- Create Test product: Manufacture buy + subcontractor vendor
- Create Bill material from Test product to subcontractor and set a component (Test component)
- Manually update the quantity of Test component in subcontracting location to have it in stock
- Create a purchase order to subcontractor for Test product
- Confirm the purchase order
- Validate incoming picking

With `mrp_subcontracting_skip_no_negative` installed no error will be shown in the last step, otherwise yes.

File: mrp_subcontracting_skip_no_negative/tests/test_mrp_subcontracting_skip_no_negative.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        )
+
+    def test_mrp_subcontracting_skip_no_negative(self):
+        picking_form = Form(self.env["stock.picking"])
+        picking_form.picking_type_id = self.env.ref("stock.picking_type_in")
+        picking_form.partner_id = self.subcontractor_partner1
+        with picking_form.move_ids_without_package.new() as move:
+            move.product_id = self.finished
+            move.product_uom_qty = 1
+        subcontracting_receipt = picking_form.save()
```
Comment (pedrobaeza) [inline mrp_subcontracting_skip_no_negative/tests/test_mrp_subcontracting_skip_no_negative.py:28]: What I mean is this picking is a manual one, not the PO generated one (with the MO linked to it).

Comment (victoralmau) [reply mrp_subcontracting_skip_no_negative/tests/test_mrp_subcontracting_skip_no_negative.py:28]: Creating a subcontracting with an in picking also generates the subcontracting flow (without the need for PO + confirm). This same approach is used in the tests of many modules, including `mrp_subcontracting`

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**